doc/tutorials/content/sources/resampling/resampling.cpp

```diff
@@ -2,6 +2,7 @@
 #include <pcl/io/pcd_io.h>
 #include <pcl/kdtree/kdtree_flann.h>
 #include <pcl/surface/mls.h>
+#include <pcl/surface/impl/mls.hpp>
```

Comment:
That's not how to resolve the error. Please modify `mls.h` instead

Reply:
Do you mean adding this line: `#include <pcl/surface/impl/mls.hpp>` to `pcl/surface/mls.h`?

Reply:
You need to remove the `ifdef PCL_NO_PRECOMPILE` in mls.h and add it to the template declarations in mls.hpp

Reply:
Disclaimer: did not check yet mls.h. If you feel the need to include mls.hpp, the following should be equivalent:
```cpp
#define PCL_NO_PRECOMPILE
#include <pcl/surface/mls.h>
```
if this is still causing compilation issues, then the precompilation implementation in mls.h is faulty and that's where we need to act as Kunal suggested.

Reply:
Sorry, but I later found the compilation problem originates from the lack of Qhull in my environment. After I installed Qhull and rebuild pcl, the problem went away.

Reply:
I've removed those commits related to the previous `make` problem. Now this PR only has something to do with the text part in the tutorial.